src/main/java/com/slack/kaldb/metadata/core/PersistentMutableMetadataStore.java

```diff
@@ -0,0 +1,46 @@
+package com.slack.kaldb.metadata.core;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.SettableFuture;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.slack.kaldb.metadata.zookeeper.MetadataStore;
+import org.slf4j.Logger;
+
+public abstract class PersistentMutableMetadataStore<T extends KaldbMetadata>
+    extends UpdatableCacheableMetadataStore<T> {
+
+  private final boolean updatable;
+
+  public PersistentMutableMetadataStore(
+      boolean shouldCache,
+      boolean updatable,
+      String storeFolder,
+      MetadataStore metadataStore,
+      MetadataSerializer<T> metadataSerializer,
+      Logger logger)
```

Comment:
Why are we passing in the logger instead of initializing it in this class? This makes invoking logging inconsistent (ie, some cases you need to do `LOG.error()` vs `logger.error()`.

Reply:
All of the log stores would use the same base class. So, it would be very confusing to understand which class is taking an action if we used the same base class logger. We pass in the child class logger here it's obvious which store is logging making the change. So, all SnapshotStore api calls will be logged under it's name as opposed to KaldbMetadataStore, which is also called by SearchStore and other stores in the future.